File: 0387_First_Unique_Character_in_a_String/Solution.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """

        charCount = dict()
        
        for c in s:
        	if c in charCount:
        		charCount[c] += 1
        	else:
        		charCount[c] = 1
        for i in range(len(charCount)):
        	if charCount[s[i]] == 1:
        		return i
        return -1

File: 0387_First_Unique_Character_in_a_String/Solution.java
public class Solution {
    public static int firstUniqChar(String s) {
        if (s.isEmpty()) {
            return -1;
        }

        int[] charCount = new int[26];

        for (int i = 0; i < s.length(); ++i) {
            char c = s.charAt(i);
            charCount[c - 'a']++;
        }

        for (int i = 0; i < s.length(); ++i) {
            char c = s.charAt(i);
            if (charCount[c - 'a'] == 1) {
                return i;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        System.out.println(firstUniqChar("aadadaad"));
    }
}